src/Parser/Core/Modules/Features/EncounterStats.js

```diff
@@ -0,0 +1,243 @@
+            />
+            { item.name }
+          </ItemLink>
+        </div>
+      </div>
+    </div>;
+  }
+
+  render() {
+    const rows = [15, 30, 45, 60, 75, 90, 100];
+
+    if (this.state.loaded) {
+      return (
+        <div style={{ border: 0, marginTop: 40 }}>
+          <div className="panel-heading results btn-link selected" style={{ padding: 20, marginTop: 60 }}>
+            <h2>This shows statistics of this fight from the top { this.LIMIT } logs, ranked by { this.metric.toLocaleUpperCase() }</h2>
+          </div>
+          <div className="flex-main">
+            <div className="row">
+              <div className="col-md-6">
+                <div className="flex-main">
+                  <div className="panel-heading" style={{ boxShadow: 'none', borderBottom: 0 }}>
+                    <h2>Most used Legendaries</h2>
+                  </div>
+                  <div className="row" style={{ paddingLeft: 20 }}>
+                    {this.state.mostUsedLegendaries.map((legendary, index) =>
+                      this.singleItem(legendary, index)
+                    )}
+                  </div>
+                </div>
+                <div className="flex-main">
+                  <div className="panel-heading" style={{ boxShadow: 'none', borderBottom: 0, marginTop: 20 }}>
+                    <h2>Most used Trinkets</h2>
+                  </div>
+                  <div className="row" style={{ paddingLeft: 20 }}>
+                    {this.state.mostUsedTrinkets.map((trinket, index) =>
+                      this.singleItem(trinket, index)
+                    )}
+                  </div>
+                </div>
+              </div>
+              <div className="col-md-6">
+                <div className="panel-heading" style={{ boxShadow: 'none', borderBottom: 0 }}>
+                  <h2>Most used Talents</h2>
+                </div>
+                {this.state.mostUsedTalents.map((row, index) => 
+                  <div className="row" key={index} style={{ marginBottom: 15, paddingLeft: 20 }}>
+                    <div className="col-md-1" style={{ lineHeight: '3em', textAlign: 'right'}}>{rows[index]}</div>
+                    {Object.keys(row).sort((a,b) => {return row[b]-row[a];}).map((talent, talentIndex) => 
+                      <div key={talentIndex} className="col-md-2" style={{ textAlign: 'center' }}>
+                        <SpellLink id={parseInt(talent, 10)}>
+                          <SpellIcon style={{ width: '3em', height: '3em' }} id={parseInt(talent, 10)} noLink />
+                        </SpellLink>
+                        <span style={{ textAlign: 'center', display: 'block' }}>{formatPercentage(row[talent] / this.LIMIT, 0)}%</span>
+                      </div>
+                    )}
+                  </div>
+                )}
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    } else {
+      this.load();
```

Comment:
You should never do any actions such as querying an API from the render method as things might be re-rendered multiple times a second which could trigger the query several times. A better place for this would be in the constructor (which is preferred over componentWillMount in ES6 classes).